sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -452,6 +452,7 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
     if (scope != null) {
       if (transaction.getRequest() == null) {
         transaction.setRequest(scope.getRequest());
+        transaction.setUser(scope.getUser());
```

Comment:
this should have its own `if` block and not within `transaction.getRequest() == null`